### PR TITLE
Audits `XrpPayment` Protobuf Wrapper

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds additional logic validations and descriptive error throwing to the class `XrpPayment`.

### Context of Change

Another component of our attempt to make protobuf wrapper classes as safe and informative as possible. An SDK safety net.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Before / After

`XrpPayment` performs logic/error checking, based on the business logic in the docs.

## Test Plan

New tests added to confirm that errors are correctly thrown. Existing tests were edited to properly account for the new errors. CI passes.

## Future Tasks
This is part of a series, with all subtasks listed here: https://app.asana.com/0/1188384817180008/1193021408531788/f

### CODE REVIEW

File: src/XRP/model/xrp-payment.ts [hunk 1]
```diff
@@ -18,20 +19,23 @@ export default class XrpPayment {
...
-    if (!amount) {
-      return undefined // amount is required
+    if (!paymentAmountValue) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Payment protobuf is missing required `amount` field.',
+      )
     }
+    const amount = XrpCurrencyAmount.from(paymentAmountValue)
+    // For non-XRP amounts, the nested field names in `amount` are lower-case.
```
Comment (amiecorso) [inline src/XRP/model/xrp-payment.ts:31]: I think we can omit checks for all of these lower-case situations.  i'm not sure if there are any similar checks implemented in `XrpCurrencyAmount`, but if we one day decide to add them they'd probably go in that class anyway.

File: src/XRP/model/xrp-payment.ts [hunk 2]
```diff
@@ -60,14 +58,29 @@ export default class XrpPayment {
...
-    if (paymentSendMaxValue && !sendMax) {
-      return undefined
+    if (sendMax && amount.drops) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Payment protobuf `sendMax` field should be omitted for XRP-to-XRP interactions.',
+      )
     }
+    // For non-XRP amounts, the nested field names in `sendMax` MUST be lower-case.
+    // `sendMax` must be supplied for cross-currency/cross-issue payments.
```
Comment (amiecorso) [inline src/XRP/model/xrp-payment.ts:78]: I believe you can implement this check in much the same way you did the `path` and XRP-to-XRP payments... I think we'd just ask about `amount.issuedCurrency` or whatever the equivalent is, && sendMax.  I think that any payment that's not XRP-to-XRP requires a sendMax... pretty sure about that.  We can always investigate and change later if need be.

**Review by amiecorso — APPROVED** [review]
really, really nice.  see inline notes about omitting case tests and potentially adding the sendMax check for issuedCurrency payments if it works out.  approving now

File: src/XRP/model/xrp-payment.ts (hunk 1 above)
Comment (mvadari) [reply src/XRP/model/xrp-payment.ts:31]: `XrpCurrencyAmount` and `XrpIssuedCurrency` don't specify any checks like this that need to be implemented. 

I'll just leave in the comments for now, just so it's noted, without a TODO or anything.

File: src/XRP/model/xrp-payment.ts [hunk 3]
```diff
@@ -18,20 +19,23 @@ export default class XrpPayment {
...
-    payment: Payment,
-    xrplNetwork: XrplNetwork,
-  ): XrpPayment | undefined {
+  public static from(payment: Payment, xrplNetwork: XrplNetwork): XrpPayment {
     const paymentAmountValue = payment.getAmount()?.getValue()
-    const amount =
-      paymentAmountValue && XrpCurrencyAmount.from(paymentAmountValue)
-    if (!amount) {
-      return undefined // amount is required
+    if (paymentAmountValue == undefined) {
```
Comment (tedkalaw) [inline src/XRP/model/xrp-payment.ts:24]: `===` pl0x

**Review by tedkalaw — CHANGES_REQUESTED** [review]
looks good; let's just change to `===`. will add a task to track us adding this lint rule

File: src/XRP/model/xrp-payment.ts [hunk 4]
```diff
@@ -60,14 +58,34 @@ export default class XrpPayment {
...
-    if (paymentSendMaxValue && !sendMax) {
-      return undefined
+    if (sendMax != undefined && amount.drops != undefined) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Payment protobuf `sendMax` field should be omitted for XRP-to-XRP interactions.',
+      )
     }
+    // For non-XRP amounts, the nested field names in `sendMax` MUST be lower-case.
+    if (sendMax == undefined && amount.issuedCurrency != undefined) {
```
Comment (tedkalaw) [inline src/XRP/model/xrp-payment.ts:78]: `===`/`!==`

**Review by tedkalaw — COMMENTED** [review]
one more place

**Review by tedkalaw — APPROVED** [review]

**Pull request merged**